include/navigation/perfect_memory_store_hog.h

```diff
@@ -78,25 +80,26 @@ class HOG
     }
 
     // Calculate difference between memory and snapshot with index
-    float calcSnapshotDifference(const cv::Mat &image, const cv::Mat &imageMask,
-                                 size_t snapshot, const cv::Mat &) const
+    float calcSnapshotDifference(const cv::Mat &image,
+                                 const ImgProc::Mask &imageMask,
+                                 size_t snapshot,
+                                 const ImgProc::Mask &) const
     {
         BOB_ASSERT(imageMask.empty());
 
+        /*
+         * Workaround for a really longstanding gcc bug:
+         *      https://gcc.gnu.org/bugzilla/show_bug.cgi?id=27557
+         */
+        static thread_local typename Differencer::template Internal<std::vector<float>> differencer;
+
```

Comment:
This is definitely for another PR but, with all these OpenMP hacks (before we even start on Windows), might it not be worth just using ``std::thread``? I parallelised some slow bits of GeNN the other day and it was _slightly_ more painful than OpenMP but at least explicit as to what was going on....

Reply:
Another option would be Intel's tbb library which looks really nice: https://software.intel.com/content/www/us/en/develop/documentation/onetbb-documentation/top.html

Plus, it's a dependency for OpenCV so users will have it installed anyway.